daemons/compute/server/servers/server_default.go

```diff
@@ -247,7 +256,136 @@ func (s *defaultServer) Create(ctx context.Context, req *pb.DataMovementCreateRe
+	} else {
+		response, err = s.createRsyncNodeDataMovement(ctx, req, userId, groupId, computeMountInfo)
+	}
+	return response, err
+}
+
+func getDirectiveIndexFromClientMount(object *dwsv1alpha1.ClientMount) (string, error) {
+	// Find the DW index for our work.
+	labels := object.GetLabels()
+	if labels == nil {
+		return "", fmt.Errorf("Unable to find labels on compute ClientMount, namespaces=%s, name=%s", object.Namespace, object.Name)
+	}
+
+	dwIndex, found := labels[nnfv1alpha1.DirectiveIndexLabel]
+	if !found {
+		return "", fmt.Errorf("Unable to find directive index label on compute ClientMount, namespace=%s name=%s", object.Namespace, object.Name)
+	}
+
+	return dwIndex, nil
+}
+
+func (s *defaultServer) createNnfDataMovement(ctx context.Context, req *pb.DataMovementCreateRequest, userId uint32, groupId uint32, computeClientMount *dwsv1alpha1.ClientMount) (*pb.DataMovementCreateResponse, error) {
+
+	var dwIndex string
+	if dw, err := getDirectiveIndexFromClientMount(computeClientMount); err != nil {
+		return &pb.DataMovementCreateResponse{
+			Status:  pb.DataMovementCreateResponse_FAILED,
+			Message: err.Error(),
+		}, nil
+	} else {
+		dwIndex = dw
+	}
+
+	lustrefs, err := s.findDestinationLustreFilesystem(ctx, req.GetDestination())
+	if err != nil {
+		return &pb.DataMovementCreateResponse{
+			Status:  pb.DataMovementCreateResponse_FAILED,
+			Message: err.Error(),
+		}, nil
+	}
+
+	// We don't have the actual NnfDataMovement parent available, but we know the name
+	// and the namespace because they will match the workflow's name and namespace.
+	parentDm := &nnfv1alpha1.NnfDataMovement{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      req.Workflow,
+			Namespace: req.Namespace,
+		},
+	}
+
+	// We don't have the actual source NnfStorage available, but we know
+	// the name and the namespace because they will match the workflow's
+	// name, with the directive index, and namespace.
+	source := &nnfv1alpha1.NnfStorage{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-%s", req.Workflow, dwIndex),
+			Namespace: req.Namespace,
+		},
+	}
+
+	dm := &nnfv1alpha1.NnfDataMovement{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "", // below
+			// Use the compute's namespace.
+			Namespace: s.name,
+			Labels: map[string]string{
+				dmctrl.InitiatorLabel: s.name,
+			},
+		},
+		Spec: nnfv1alpha1.NnfDataMovementSpec{
+			Destination: &nnfv1alpha1.NnfDataMovementSpecSourceDestination{
+				Path: req.GetDestination(),
+				Storage: &corev1.ObjectReference{
+					Kind:      reflect.TypeOf(*lustrefs).Name(),
+					Namespace: lustrefs.Namespace,
+					Name:      lustrefs.Name,
+				},
+			},
+			Source: &nnfv1alpha1.NnfDataMovementSpecSourceDestination{
+				Path: "/",
+				Storage: &corev1.ObjectReference{
+					Kind:      reflect.TypeOf(*source).Name(),
+					Namespace: source.Namespace,
+					Name:      source.Name,
+				},
+			},
+			UserId:  userId,
+			GroupId: groupId,
+		},
+	}
+
+	dwsv1alpha1.AddOwnerLabels(dm, parentDm)
```

Comment:
Is anything using these owner labels? The datamovement controller isn't setup to delete children with the datamovement type. Should the owner be the workflow instead?